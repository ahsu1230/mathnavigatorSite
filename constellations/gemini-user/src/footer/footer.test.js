import React from "react";
import Enzyme, { shallow } from "enzyme";
import { HashRouter as Router } from "react-router-dom";
import Footer from "./footer.js";
import renderer from "react-test-renderer";

describe("test", () => {
    const component = shallow(<Footer />);

    test("renders", () => {
        expect(component.exists()).toBe(true);
    });

    test("shows links", () => {
        var ul = component.find("ul");
        expect(ul.exists()).toBe(true);
        expect(ul.children().length).toBe(4);
    });

    test("snapshot", () => {
        const tree = renderer
            .create(
                <Router>
                    <Footer />
                </Router>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
