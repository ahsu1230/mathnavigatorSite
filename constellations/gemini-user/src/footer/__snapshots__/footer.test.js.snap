// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  id="view-footer"
>
  <div
    id="view-footer-container"
  >
    <ul>
      <li>
        <a
          href="#/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="#/programs"
          onClick={[Function]}
        >
          Programs
        </a>
      </li>
      <li>
        <a
          href="#/student-achievements"
          onClick={[Function]}
        >
          Accomplishments
        </a>
      </li>
      <li>
        <a
          href="#/register"
          onClick={[Function]}
        >
          Register
        </a>
      </li>
    </ul>
    <div
      className="logo-container"
    >
      <a
        href="#/"
        id="footer-logo"
        onClick={[Function]}
      >
        <img
          src="test-file-stub"
        />
        <h1
          className="logo"
        />
      </a>
      <h2>
        Montgomery County, MD
      </h2>
      <p>
        Program sessions are held off-school hours at local public schools.
        <br />
        Math Navigator is not affiliated with those schools.
      </p>
      <p>
        For any questions, please email us at
         
        <u>
          andymathnavigator@gmail.com
        </u>
        .
      </p>
    </div>
  </div>
</div>
`;
